2020/day3/solution.py

```diff
@@ -0,0 +1,43 @@
+file = open('input.txt')
+
+class grid:
+    
+    def __init__(self, lines):
+        self.lines = list(lines)
```

Comment:
```suggestion
        self.lines = lines
```
The `list` call makes a copy of the list. However, you don't mutate `self.lines` in the function so I wouldn't bother. Style-wise, making a copy can be warranted if your class will mutate its input, but generally I try to avoid that in favor of either:
- having the class not mutate the list passed in
- if the class will mutate the list, have it create it itself in its constructor (possibly via list comprehension over some actual argument e.g.)
Leaning on copies leads to the C++ problem where you have allocations all over the place because it was too hard to get a reference safely since you weren't tracking other uses (shared mutability gone amok); immutability & clear ownership boundaries both help here.